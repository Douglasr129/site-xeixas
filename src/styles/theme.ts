/* eslint-disable quotes */
export const theme = {
	colors: {
		primary: '#7395B8',
		secondary: '#dc143c',
		action: '#81AE85',
		white: '#FFFFFF',
		medium: '#E3DEFF',
		dark: '#161421',
	},
	actionsColors: {
		primary: '#4B9DB8',
		secondary: '#AD1E02',
		action: '#4CAD70',
		medium: '#B77BB8',
		dark: '#10009E',
	},
	fonts: {
		family: {
			default: "'Open Sans', sans-serif",
			secundary: "'Montserrat',sans-serif",
		},
		sizes: {
			xsmall: '0.8rem',
			small: '1.6rem',
			medium: '2.4rem',
			large: '3.2rem',
			xlarge: '4.0rem',
			xxlarge: '4.8rem',
			huge: '5.6rem',
			xhuge: '6.4rem',
		},
	},
	medias: {
		mobileS: '(max-width: 320px)',
		mobileM: '(max-width: 375px)',
		mobileL: '(max-width: 425px)',
		tablet: '(max-width: 768px)',
		laptop: '(max-width: 1024px)',
		laptopL: '(max-width: 1440px)',
		desktop: '(max-width: 2560px)',
	},
	spacings: {
		xsmall: '0.8rem',
		small: '1.6rem',
		medium: '2.4rem',
		large: '3.2rem',
		xlarge: '4.0rem',
		xxlarge: '4.8rem',
		huge: '5.6rem',
		xhuge: '6.4rem',
	},
	paddings: {
		xsmall: '0.8rem',
		small: '1.6rem',
		medium: '2.4rem',
		large: '3.2rem',
		xlarge: '4.0rem',
		xxlarge: '4.8rem',
		huge: '5.6rem',
		xhuge: '6.4rem',
	},
};
