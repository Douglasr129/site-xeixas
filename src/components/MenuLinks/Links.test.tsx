import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { theme } from '../../styles/theme';
import { Links } from '.';
import React from 'react';

describe('<Links />', () => {
	it('should render a link', () => {
		renderTheme(<Links link="http://localhost">Children</Links>);
		expect(screen.getByRole('link', { name: 'Children' })).toHaveAttribute(
			'target',
			'_self',
		);
	});

	it('should render a internal link', () => {
		renderTheme(<Links link="/localhost">Children</Links>);
		expect(screen.getByRole('link', { name: 'Children' })).toHaveAttribute(
			'target',
			'_self',
		);
	});

	it('should render open in a new tab', () => {
		renderTheme(
			<Links link="http://localhost" newTab={true}>
				Children
			</Links>,
		);
		expect(screen.getByRole('link', { name: 'Children' })).toHaveAttribute(
			'target',
			'_blank',
		);
	});
	it('should render in dark mode', () => {
		renderTheme(
			<Links darkMode={true} link="http://localhost">
				Children
			</Links>,
		);
		const textComponent = screen.getByText('Children');
		expect(textComponent).toHaveStyleRule('color', theme.colors.white);
	});
	/* 	it('should render in dark mode', () => {
		const { container } = renderTheme(
			<Links link="http://localhost" darkMode={true}>
				Children
			</Links>,
		);
		console.log('Style:', container.firstChild);
		expect(container.firstChild).toHaveStyle({ color: theme.colors.white });
	}); */
});
