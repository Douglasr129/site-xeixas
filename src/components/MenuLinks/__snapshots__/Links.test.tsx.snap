// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Links /> should have snepshot 1`] = `
.c0 {
  padding: 0.8rem;
  margin: 0.8rem;
  text-decoration: none;
  color: #FFFFFF;
  text-align: center;
}

.c0:hover {
  border-bottom: solid 0.2rem #dc143c;
}

<a
  class="c0"
  href="http://localhost"
  target="_self"
>
  Children
</a>
`;
